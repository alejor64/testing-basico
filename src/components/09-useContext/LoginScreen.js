import React, { useContext } from 'react'
import { UserContext } from './UserContext'

const LoginScreen = () => {

    const {setUser} = useContext(UserContext)

    const object = {
        id: 123,
        name: 'Alejandro'
    }

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={() => setUser(object)}
            >
                Login
            </button>
        </div>
    )
}

export default LoginScreen
