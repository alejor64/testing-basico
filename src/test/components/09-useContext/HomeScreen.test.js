import React from 'react'
import '@testing-library/jest-dom'
import {mount} from 'enzyme'
import HomeScreen from '../../../components/09-useContext/HomeScreen'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('Pruebas en el componente <HomeScreen />', () => {

    const user = {
        name: 'Alejandro',
        email: 'alejoruizs@hotmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    )

    test('Debe de mostrar el compoenente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
})
