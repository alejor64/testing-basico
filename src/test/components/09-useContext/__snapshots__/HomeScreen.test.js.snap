// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HomeScreen /> Debe de mostrar el compoenente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Alejandro",
   "email": "alejoruizs@hotmail.com"
}
  </pre>
</div>
`;
